tudatBundle/pagmo2/include/pagmo/algorithms/giACOmo.hpp

```diff
@@ -377,13 +342,25 @@ class gi_aco_mo
                                             " Error: the initial population does not have at least m_ker feasible individuals to be stored in the solution archive ");
 
                             }
+
+
                             //I initialize the solution archive (SA): in order to do this I store the vectors generated in the first generation
                             //by taking into account their penalty values. In this way, the first vector in the SA (i.e., the first row in which
                             //penalty value, variables, objective functions values, equality constraints values, inequality constraints values are
                             //stored) represents the best one (i.e., the one that has the smallest penalty function value among the individuals of
                             //that generation), whereas the last vector represents the worst.
                             for (decltype(m_ker) i=0u; i<m_ker; ++i){
-                                SA[i]={penalties[sorted_penalties[i]], X[sort_list[i]], fit[sort_list[i]]};
+                                SA[i][0] = penalties[sorted_penalties[i]];
+
+                                for (decltype(dim) J=1; J<=dim; ++J){
+                                    double A = X[sort_list[i]][J];
+                                    SA[i][J] = A;
+
+                                }
+                                for (decltype(NALL) J=dim+1; J<=dim+NALL; ++J){
+                                    double B = fit[sort_list[i]][J];
+                                    SA[i][J] = B;
```

Comment:
The definition of A and B is redundant, I forgot to erase them, I was using them just to figure out a compiler error that I was having. In the next commit I will remove them.